src/main.rs

```diff
@@ -0,0 +1,83 @@
+use bytes::{ Bytes, BytesMut};
+use std::net::SocketAddr;
+use tokio::net::{TcpListener, TcpStream};
+use tokio_stream::StreamExt;
+use tokio_util::codec::Framed;
+use futures::SinkExt;
+use anyhow::Result;
+use clap::App;
+
+pub mod tokio_netstring;
+use tokio_netstring::NetstringCodec;
+
+#[tokio::main]
+async fn main() -> anyhow::Result<()> {
+    let matches = App::new("simple").args_from_usage("-s --server 'Server mode'").get_matches();
+    let addr = "127.0.0.1:20000".to_string();
+
+    if matches.is_present("server") {
+        let socket = TcpListener::bind(&addr).await?;
+        loop {
+            let (connection, clientaddr) = socket.accept().await?;
```

Comment:
This will kill the server on failure. Should it be an error print instead?

Reply:
Technically true, practically its hard to fail to accept connection, it would happen either on some RO systems or low on memory. However, good catch, i will fix it somehow.